### PR TITLE
Fix remaining GPU tests

### PR DESCRIPTION
**Pull request by tgaddair**

### CODE REVIEW

**Comment by justinxzhao** [comment]
It looks like the vast majority of the remaining failures are coming from tab transformer:

```
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-concat-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-concat-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-concat-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-concat-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-sum-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-sum-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-sum-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-None-sum-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-concat-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-concat-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-concat-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-concat-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-sum-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-sum-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-sum-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[None-fc_layers1-sum-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-concat-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-None-sum-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-concat-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[64-fc_layers1-sum-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-concat-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-None-sum-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-concat-2-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-1-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-1-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-1-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-1-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-1-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-1-feature_list5]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-2-feature_list0]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-2-feature_list1]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-2-feature_list2]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-2-feature_list3]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-2-feature_list4]
FAILED tests/ludwig/combiners/test_combiners.py::test_tabtransformer_combiner[add-fc_layers1-sum-2-feature_list5]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_dense_encoder[4-vocab0]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_dense_encoder[4-vocab1]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_dense_encoder[6-vocab0]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_dense_encoder[6-vocab1]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_dense_encoder[10-vocab0]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_dense_encoder[10-vocab1]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_sparse_encoder[vocab0]
FAILED tests/ludwig/encoders/test_category_encoders.py::test_categorical_sparse_encoder[vocab1]
FAILED tests/ludwig/features/test_bag_feature.py::test_bag_input_feature[embed]
```

@ShreyaR FYI

**Comment by anmshkmr** [comment]
Looking into the failures

**Comment by anmshkmr** [comment]
The `TabTransformerCombiner` forward method takes in the inputFeature and does higher level transform on the feature itself. This fix should take care of one type of failures. Will look into the others next.

File: ludwig/combiners/combiners.py [hunk 1]
```diff
@@ -810,7 +811,7 @@ def forward(
             hidden = torch.permute(hidden, (1, 0, 2))  # bs, num_eo, h
 
             if self.embed_input_feature_name:
-                i_f_names_idcs = torch.reshape(torch.arange(0, len(embeddable_encoder_outputs)), [-1, 1])
+                i_f_names_idcs = torch.reshape(torch.arange(0, len(embeddable_encoder_outputs)).to(DEVICE), [-1, 1])
```
Comment (tgaddair) [inline ludwig/combiners/combiners.py:814]: @anmshkmr the issue here seems to be that these tensors are being constructed at runtime. Instead, we should aim to construct them as properties of the module during initialization, that way they will be placed on the correct device when we call `model.cuda()`.

**Pull request merged**